rust/otap-dataflow/crates/otap/src/syslog_cef_receiver.rs

```diff
@@ -110,43 +123,63 @@ pub static SYSLOG_CEF_RECEIVER: ReceiverFactory<OtapPdata> = ReceiverFactory {
+                        biased; // Prioritize control messages over data
 
                         // Process incoming control messages.
                         ctrl_msg = ctrl_chan.recv() => {
                             match ctrl_msg {
-                                Ok(NodeControlMsg::Shutdown {..}) => {
-                                // ToDo: Add proper deadline function
-                                break;
-                                },
-                            Err(e) => {
-                                return Err(Error::ChannelRecvError(e));
+                                Ok(NodeControlMsg::Shutdown { .. }) => {
+                                    if let Some(h) = telemetry_timer.take() {
+                                        let _ = h.cancel().await;
+                                    }
+                                    break;
+                                }
+                                Ok(NodeControlMsg::CollectTelemetry { mut metrics_reporter }) => {
+                                    let mut guard = self.metrics.borrow_mut();
+                                    let _ = metrics_reporter.report(&mut guard);
                                 }
-                            _ => {
-                                // ToDo: Handle other control messages if needed
+                                Err(e) => {
+                                    return Err(Error::ChannelRecvError(e));
+                                }
+                                _ => {
+                                    // ToDo: Handle other control messages if needed
                                 }
                             }
                         }
 
+
                         // Process incoming TCP connections.
                         accept_result = listener.accept() => {
                             match accept_result {
                                 Ok((socket, _peer_addr)) => {
-                                    // Clone the effect handler so the spawned task can send messages.
+                                    // Track active connections
+                                    {
+                                        let mut m = self.metrics.borrow_mut();
+                                        m.tcp_connections_active.inc();
+                                    }
+
+                                    // Clone the effect handler and metrics so the spawned task can send messages and record telemetry.
                                     let effect_handler = effect_handler.clone();
+                                    let metrics = self.metrics.clone();
+
                                     // Spawn a task to handle the connection.
-                                    // ToDo should this be abstracted and exposed a method in the effect handler?
-                                    _ = tokio::task::spawn_local(async move {
+                                    drop(tokio::task::spawn_local(async move {
```

Comment:
I think it's best to leave this as is. We want to keep the TODO comment and intentionally use `_ = tokio::task::spawn_local(async move { ... })` instead of dropping the handle. I plan to use that handle later when working on the graceful shutdown feature.

In general, I'd recommend keeping changes focused on the purpose of this PR and avoiding unrelated modifications.